### DISCUSSION TITLE
Show message of DatasourceError in State history

### DISCUSSION THREAD

**Discussion by Yogu** [grafana/grafana#62672] [Alerting]
If you have an alert that cycles between "Normal" and "DatasourceError", it is currently very hard to find out the exact reason. The error message is only displayed while the error occurs. If it heals itself quickly and takes a while to go back to error, you would need to constantly refresh the alert page or act very quickly when it errors to find out the cause.

I propose to include the error message of the "Error" entries in the "State history" of the alert. If this would take too much disk space, it would already help to show the error message of the last Error entry.

![image](https://user-images.githubusercontent.com/395810/216037051-ed8b1e81-33ca-484d-9d70-9b0ed45fe7bd.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82861. Feel free to continue the discussion around this there. Thank you!
